audformat/core/utils.py

```diff
@@ -269,6 +312,25 @@ def concat(
             )
         columns_reindex[column.name][column.index] = column
 
+    # Apply custom aggregation function
+    # on collected overlapping data
+    # (no overlapping data is collected
+    #  when no aggregation function is provided)
+    if len(overlapping_values) > 0:
+        for column in overlapping_values:
+            df = pd.concat(
+                overlapping_values[column],
+                axis=1,
+                ignore_index=True,
+            )
+            dtype = columns_reindex[column].dtype
+            columns_reindex[column] = df.apply(aggregate_function, axis=1)
+            # Restore the original dtype if possible
+            try:
+                columns_reindex[column] = columns_reindex[column].astype(dtype)
+            except (TypeError, ValueError):
+                pass
```

Comment:
Could it make sense to raise here to avoid the `dtype` silently changes?

Reply:
Not really. E.g. when we decide to aggregate the values inside a tuple, the dtype has to change.

Note, there was also an error and I had to rewrite this section as:

```python
            y = df.apply(aggregate_function, axis=1)
            # Restore the original dtype if possible
            try:
                y = y.astype(dtype)
            except (TypeError, ValueError):
                columns_reindex[column] = columns_reindex[column].astype(
                    y.dtype
                )
            columns_reindex[column].loc[y.index] = y
```